virtme/commands/run.py

```diff
@@ -487,17 +492,22 @@ def find_kernel_and_mods(arch, args) -> Kernel:
                 args.kimg = None
 
     if args.kimg is not None:
-        img_name = arch.img_name()
+        img_names = arch.img_name()
 
-        # Try to resolve kimg as a kernel version first, then check if a file
-        # is provided.
-        kimg = "/usr/lib/modules/%s/%s" % (args.kimg, img_name)
-        if not os.path.exists(kimg):
+        for img_name in img_names:
+            # Try to resolve kimg as a kernel version first, then check if a file
+            # is provided.
+            kimg = "/usr/lib/modules/%s/%s" % (args.kimg, img_name)
+            if os.path.exists(kimg):
+                break
             kimg = "/boot/%s-%s" % (img_name, args.kimg)
+            if os.path.exists(kimg):
+                break
+        else:
+            kimg = args.kimg
             if not os.path.exists(kimg):
-                kimg = args.kimg
-                if not os.path.exists(kimg):
-                    arg_fail("%s does not exist" % args.kimg)
+                arg_fail("%s does not exist" % args.kimg)
+
         kver = get_kernel_version(img_name, kimg)
```

Comment:
We probably need to move get_kernel_version() inside the for img_name loop as well.